### PR TITLE
Add component and redux store

### PR DESCRIPTION
**Pull request by bolah2009**
### Add components and `Redux` store according to the following requirements:

1. Prepare the directory structure
    - Create a `components` directory in the `src` folder for all the presentational components.
    - Create a `containers` directory in the `src` folder for all the container components.
    - Create a `actions` directory with the `index.js` file where you'll define all the Redux actions.
    - Create a `reducers` directory with two files: `books.js` and `index.js`. The former will store the functions responsible for actions on Books, and the latter should combine all the reducers into one. For now, it will only combine the `books.js` reducer.
    - Documentation: [Combine Reducers](https://redux.js.org/recipes/structuring-reducers/using-combinereducers)
2. Implement the **App** component
    - It should only wrap `BooksList` and `BooksForm` in a `div` selector and display it.
    - Attach this component to render the function in the main `index.js` file.
3. Implement the **BooksList** component
    - It should display the books as an html table.
    - Each row should contain Book ID, title and category.
    - It should connect to the Redux store and fetch the books from the state.
4. Implement the **Book** component
    - It should accept a book object as a prop.
    - It should represent a single row in the Books table.
5. Implement the **BooksForm** component
    - It should render a very simple form for adding books. The form should contain a book title input, a select box with categories and a submit button.
    - Define a constant with a collection containing the book categories: `["Action", "Biography", "History", "Horror", "Kids", "Learning", "Sci-Fi"]`.
6. Define the Redux actions and reducers
    - Implement the `books.js` reducer with `CREATE_BOOK` and `REMOVE_BOOK` actions. Remember, your state can not be mutated!
    - Define the `CREATE_BOOK` and `REMOVE_BOOK` actions in `actions/index.js`. They should accept a `book` object as an argument.
7. Initialize the Redux store
    - Create the initial state object in `src/index.js`. It may contain just a few books with different titles and categories.
    - Each book in the initial state should have a unique ID.
    - Render the `<App/>` component wrapped in `<Provider>` and pass the store down to the components.

### Deploy Preview:

https://deploy-preview-2--bolah-react-bookstore.netlify.com/

### Screenshot

image

### CODE REVIEW

File: src/components/BooksForm.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  return (
+    <select name="categories" id="category">
+      <option key="placeholder" value="">
+        Category
+      </option>
+      {options}
+    </select>
+  );
+};
```
Comment (SunnyAsar) [inline src/components/BooksForm.jsx]: These two things could be one. and usually you want the first category item to be the first in a select/option.

File: src/containers/BooksList.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import Book from '../components/Book';
+import styles from '../css/modules/BooksList.module.css';
+
+const BookList = ({ books }) => {
```
Comment (SunnyAsar) [inline src/containers/BooksList.jsx:7]: you'll need to hold state for the form, this should have been a class component.

**Review by SunnyAsar — APPROVED** [review]
# Hi @bolah2009 👋 
### Good job here.
 - I have made a comment below, there is nothing deal breaking but keep that in mind in the next milestone.
- There really is no need to put the whole milestone or project requirement in you PR instead, just a little summary of what has been done in the PR is enough.

## You May Pass 👍

File: src/components/BooksForm.jsx (hunk 1 above)
Comment (bolah2009) [inline src/components/BooksForm.jsx:21]: I think it's a personal preference for me. I don't want an already chosen option so the default is an empty value.

File: src/containers/BooksList.jsx (hunk 2 above)
Comment (bolah2009) [reply src/containers/BooksList.jsx:7]: Yes, I did that in the next milestone #3 . I think using a functional component for a stateless component is better. Also, my eslint configuration won't allow it.

**Pull request merged**